src/dk/aau/cs/model/CPN/ColoredTimeInterval.java

```diff
@@ -0,0 +1,161 @@
+            } else
+                throw new RuntimeException("A constant which was not declared was used in an time interval of an arc.");
+        }
+
+        Bound upperBound = null;
+        if (upperBoundAsString.equals("inf"))
+            upperBound = Bound.Infinity;
+        else {
+            try {
+                int intBound = Integer.parseInt(upperBoundAsString);
+                upperBound = new IntBound(intBound);
+            } catch (NumberFormatException e) {
+                if (constants.containsConstantByName(upperBoundAsString)) {
+                    upperBound = new ConstantBound(constants.getConstantByName(upperBoundAsString));
+                } else
+                    throw new RuntimeException(
+                            "A constant which was not declared was used in an time interval of an arc.");
+            }
+        }
+
+        return new ColoredTimeInterval(leftBracket.equals("[") ? true : false,
+                lowerBound, upperBound, rightBracket.equals("]") ? true : false, color);
+    }
+
+
+    @Override
+   public String toString() {
+        String print = "";
+        if (color != null) {
+            if (color.getTuple() != null) {
+                for (Color color1 : color.getTuple()) {
+                    print += color1.getColorName() + ", ";
+                }
+                print.substring(0, print.length()-2);
+                print += " \u2192 " + super.toString();
+                return print;
+            } else {
+                print += color.getColorName() + " \u2192 " + super.toString();
+                return print;
+            }
+        } else {
+            print += super.toString();
+            return print;
+        }
+    }
+
+    public String getInterval() {
+        return super.toString();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        ColoredTimeInterval cti;
+        if (o instanceof ColoredTimeInterval) {
+            cti = (ColoredTimeInterval)o;
+        } else {
+            return false;
+        }
+
+        if (!(cti.isLowerIncluded == this.isLowerIncluded))
+            return false;
+
+        if (cti.lowerBound().value() != this.lowerBound().value())
+            return false;
+
+        if (cti.upperBound().value() != this.upperBound().value())
+            return false;
+
+        if (cti.isUpperIncluded != this.isUpperIncluded)
+            return false;
+
+        if (!cti.color.equals(this.color))
+            return false;
+
+        return true;
+    }
+
+    public boolean equalsOnlyColor(Object o) {
+        ColoredTimeInterval cti;
+        if (o instanceof ColoredTimeInterval) {
+            cti = (ColoredTimeInterval)o;
+        } else {
+            return false;
+        }
+        if (!cti.color.equals(this.color))
+            return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Could properly by handles better by using Objects.hashCode(lowerBound(), upperBound(), color).
Also check that this is not used anywhere in a hash map, with mutations.